## Assignment: Homework 3 M Section
## Name: Kiran Venkata Palla
## Net Id: kvp246

# Assignment 2:  Hypothesis testing

Follow and understand the notebook https://github.com/fedhere/PUI2015_fbianco/blob/master/HW3/effectivenes%20of%20NYC%20Post-Prison%20Employment%20Programs.ipynb
and repeat the z-test and chisq test for other data included in the paper we are working with. 
by fill in missing cells in (your own copy of) the notebook

## Test of proportions on whether the former inmates were or were not ever employed in a CEO transitional job using 2 sample z-test

NULL HYPOTHESIS: the % of former prisoners employed 3 years after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

H0: P_0 - P_1 >= 0

H1: P_0 - P_1 < 0

$\alpha$ = 0.05

In [2]:
# Importing required libraries
import numpy as np

# Alpha value, level of significane 95%
alpha = 0.05

# Proportion values of employment in test and control samples
P_0 = 52.1*0.01 
P_1 = 53.3*0.01

print "Control group proportion (P_0):", P_0
print "Test group proportion (P_1):", P_1

# Sample sizes of test and control groups
n_0 = 409
n_1 = 564

# Relative frequency of employment in test and control samples
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

# Sample proportion
sp = (P_0 * n_0 + P_1 * n_1)/(n_1 + n_0)
print "Sample proportion:", sp

# Function for computing standard deviation value to compute z-score
sp_stdev = lambda p, n: np.sqrt( p * ( 1 - p ) / n[0] +  p * ( 1 - p )/ n[1]  )

sp_stdev_3y = sp_stdev((Nt_0 + Nt_1)/(n_0 + n_1),[n_0, n_1])

print "Standard deviation:", sp_stdev_3y

# Computing the z-score
zscore = lambda p0, p1, s : (p0 - p1)/s

z_3y = zscore(P_1, P_0, sp_stdev_3y)

print "z-score:", z_3y

## Looking at the standard normal distribution tables, cumulative probability (P) value is considered as 0.6368
## This P is taken based on the aplha value of 0.05 and z-score of ~0.37011

p_3y=1-0.6368

print "p-value:", p_3y

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

report_result(p_3y,alpha)

Control group proportion (P_0): 0.521
Test group proportion (P_1): 0.533
Sample proportion: 0.527955806783
Standard deviation: 0.0324224126866
z-score: 0.370114343926
p-value: 0.3632
is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


## Computing the chi-square statistic for the hypothesis about the employment for >6 consecutive quarters

|employed for > 6 quarters |     employed   | not employed   |                   
|---------------------------|----------------|------------------|---------------------------|
| test sample               |   a  |  b    | a+b                       |
| control sample            |   c |  d    | c+d                       |
|                           |                |                  |                           |
| total                     |   a+c     |  b+d         | a+b+c+d         |

For a 2 x 2 contingency table the Chi Square statistic is calculated by the formula:

Chi-square Statistic = ((ad-bc)^2 * (a+b+c+d)) / ((a+b)(c+d)(a+c)(b+d))

In [48]:
# Category of Interest = "Employed for six or more consecutive quarters (%)"
# Performing the chi-square test on the given data

Tot_Program_Grp = 564
Tot_Control_Grp = 409

Tot_Employed_Program_Grp = 0.147 * Tot_Program_Grp
Tot_UnEmployed_Program_Grp = Tot_Program_Grp - Tot_Employed_Program_Grp

Tot_Employed_Control_Grp = 0.119 * Tot_Control_Grp
Tot_UnEmployed_Control_Grp = Tot_Control_Grp - Tot_Employed_Control_Grp

a = Tot_Employed_Program_Grp
b = Tot_UnEmployed_Program_Grp
c = Tot_Employed_Control_Grp
d = Tot_UnEmployed_Control_Grp

Ntot = a + b + c + d
expected = (a + b)*(c + d)*(a + c)*(b + d)
sample_values = [[a,b],[c,d]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


Based on the given 2x2 contingency table data, say with an $\alpha$ value of 0.05 (significance level) and for degrees of freedom = 1 ((row-1) * (col-1)), the critical value for this chi-square distribution would be 3.841. This critical value is taken from the standard chi-square distribution tables for the given values of $\alpha$ = 0.05 and df = 1.

Now based on the computed value of ~1.589, we can notice that this is lesser than the critical value of 3.841 and tends to fall under the significance level values of 0.9 and 0.05. This confirms the fact that p-value for the computed value is greater than 0.05, and based on this fact,

### We fail to reject the NULL Hypothesis that there is no statistically significant difference in the proportion of prison test and control groups with respect to employability for more than 6 consecutive quarters